packages/experience-builder-sdk/src/utils/parseComponentProps.ts

```diff
@@ -0,0 +1,207 @@
+import { isCfStyleAttribute } from '@contentful/experiences-core';
+import {
+  BoundComponentPropertyTypes,
+  BoundValue,
+  Breakpoint,
+  ComponentDefinition,
+  ComponentDefinitionVariable,
+  ComponentDefinitionVariableType,
+  ComponentTreeNode,
+  DesignValue,
+  PrimitiveValue,
+} from '@contentful/experiences-core/types';
+import { convertResolvedDesignValuesToMediaQuery } from '../hooks/useMediaQuery';
+
+const isSpecialCaseCssProp = (propName: string) => {
+  return propName === 'cfBackgroundImageUrl' || propName.startsWith('cfBackgroundImageUrl_');
+};
+
+/**
+ * The previous logic of prop mapping was too complex and mixed different ues cases together.
+ * In this function, I aim to simplify the logic by focusing on the following specific cases FOR PREVIEW/DELIVERY MODES
+ * 1) Any non `DesignValue` props should be resolved and returned as is
+ * 2) Some exceptions like `cfImageAsset` and `cfBackgroundImageUrl` (BoundValue) should be handled separately
+ * and be resolved for the default breakpoint only (cause we don't allow binding per breakpoint anyway)
+ * 3) Those DesignValue props which can be converted to CSS should be grouped and resolved into a CSS media query
+ * for each breakpoint
+ * 4) Those DesignValue props which can NOT be converted to CSS (custom design props) should be resolved dynamically
+ * for each breakpoint
+ */
+export const parseComponentProps = ({
+  mainBreakpoint,
+  componentDefinition,
+  node,
+  resolveClassNamesFromBuiltInStyles,
+  resolveCustomDesignValue,
+  resolveBoundValue,
+  resolveHyperlinkValue,
+  resolveUnboundValue,
+}: {
+  mainBreakpoint: Breakpoint;
+  node: ComponentTreeNode;
+  componentDefinition: ComponentDefinition;
+  resolveClassNamesFromBuiltInStyles: (
```

Comment:
Passing in this callback seems unnecessary as we just need the `node` and `breakpoints` for this.

Reply:
Will be tackled in https://github.com/contentful/experience-builder/pull/999